app/models/dialog_field_text_box.rb

```diff
@@ -31,7 +31,7 @@ def value_from_dialog_fields(dialog_values)
 
   def automate_output_value
     return nil if @value.nil?
-    return MiqPassword.encrypt(@value) if self.protected?
+    return MiqPassword.encrypt(@value) if self.protected? && value_is_not_already_encrypted?
```

Comment:
You should use `MiqPassword.encrypted?` instead of creating a new method.  Also, we need to use MiqPassword methods so we are protected against the encrypted string format changing in the future.

Reply:
Oh I didn't even realize this existed, nice. Will change.